libs/language-server/src/lib/ast/wrappers/value-type/valuetype-util.ts

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { assertUnreachable } from 'langium';
+
+import {
+  PrimitiveValuetypeKeyword,
+  ValuetypeDefinition,
+  ValuetypeReference,
+  isPrimitiveValuetypeKeywordLiteral,
+  isValuetypeDefinition,
+  isValuetypeDefinitionReference,
+  isValuetypeReference,
+} from '../../generated/ast';
+
+// eslint-disable-next-line import/no-cycle
+import { AtomicValuetype } from './atomic-valuetype';
+import { PrimitiveValuetypes, createPrimitiveValuetype } from './primitive';
+import { Valuetype } from './valuetype';
+
+type ValuetypeIdentifier =
+  | ValuetypeDefinition
+  | PrimitiveValuetypeKeyword
+  | 'cellRange'
+  | 'constraint'
+  | 'regex'
+  | 'transform'
+  | 'valuetypeAssignment'
+  | 'collection';
```

Comment:
Not sure whether we need this type anymore. I guess that primitive valuetypes which have no corresponding keyword in the language (e.g. `cellRange`) never need to be created by `createValuetype`.